File: Dynamic programming/leetcode_65.cpp

// Problem link ===>> https://leetcode.com/problems/valid-number/description/?envType=problem-list-v2&envId=21r4yhih

#include <bits/stdc++.h>
#include <ext/pb_ds/assoc_container.hpp>
#include <ext/pb_ds/tree_policy.hpp>
using namespace std;
using namespace __gnu_pbds;
#define ll long long
#define ld long double
#define endl '\n'
const int MOD = 1e9 + 7;
const double EPS = 1e-9;
const double PI = acos(-1.0);
template <typename T>
using ordered_set = tree<T, null_type, less<T>, rb_tree_tag, tree_order_statistics_node_update>;
template <typename K, typename V>
using ordered_map = tree<
    K, V, less<K>, rb_tree_tag, tree_order_statistics_node_update>;

class Solution
{
public:
    bool isNumber(string S)
    {
        bool num = false, exp = false, sign = false, dec = false;
        for (auto c : S)
            if (c >= '0' && c <= '9')
                num = true;
            else if (c == 'e' || c == 'E')
                if (exp || !num)
                    return false;
                else
                    exp = true, sign = false, num = false, dec = false;
            else if (c == '+' || c == '-')
                if (sign || num || dec)
                    return false;
                else
                    sign = true;
            else if (c == '.')
                if (dec || exp)
                    return false;
                else
                    dec = true;
            else
                return false;
        return num;
    }
};

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);
    
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;

        
    }
    return 0;
}

File: Dynamic programming/leetcode_65.py

# Problem link ===>> https://leetcode.com/problems/valid-number/description/?envType=problem-list-v2&envId=21r4yhih
# submission link == =>> https://leetcode.com/problems/valid-number/submissions/1754886323/?envType=problem-list-v2&envId=21r4yhih

import sys
import math
import bisect
import heapq
from collections import defaultdict, deque, OrderedDict, Counter
from functools import lru_cache, reduce
from itertools import permutations, combinations, accumulate
from typing import List, Tuple, Dict, Set, Optional, Union, Any

MOD = 10**9 + 7
INF = float('inf')
input = sys.stdin.read
print = sys.stdout.write


class Solution:
    def isNumber(self, S: str) -> bool:
        num, exp, sign, dec = False, False, False, False
        for c in S:
            if c >= '0' and c <= '9':
                num = True
            elif c == 'e' or c == 'E':
                if exp or not num:
                    return False
                else:
                    exp, num, sign, dec = True, False, False, False
            elif c == '+' or c == '-':
                if sign or num or dec:
                    return False
                else:
                    sign = True
            elif c == '.':
                if dec or exp:
                    return False
                else:
                    dec = True
            else:
                return False
        return num

def main():
    t = int(input())
    for _ in range(t):
        n = int(input())

if __name__ == "__main__":
    solution = Solution()
